h2o-core/src/main/java/hex/Model.java

```diff
@@ -1530,7 +1535,7 @@ protected Frame predictScoreImpl(Frame fr, Frame adaptFrm, String destination_ke
                                    j,
                                    customMetricFunc).doAll(names.length, Vec.T_NUM, adaptFrm);
 
-    if (computeMetrics)
+    if (computeMetrics && bs._mb != null) //metric builder can be null if training was interrupted/cancelled
```

Comment:
hmm, what other issues can there be? the model will be invalid basically - can we make sure the model will have metrics if there was a timeout? Not sure where else can it crash

Reply:
yes, ensuring `bs._mb != null` looks simple, it's just about moving one line in `BigScore.map` below. Will give it a try.

Reply:
@michalkurka I doubled checked this code: there should be no issue on timeout:
- there's no timeout on model scoring, it's handled only at model driver level.
- in case of training timeout, we throw an exception, we don't call `job.stop()`.
- only AutoML may call `job.stop()` in case of global timeout. I can ensure that `BigScore.map` always returns a `MetricBuilder` in this case, but I'm more comfortable with the null-check above: if there's no metric, the model won't appear in our leaderboard anyway.
wdyt?

Reply:
@seb-h2o Okay, I think that is reassuring.

What about models that override `predictScoreImpl`? Do they need similar handling as well?

Reply:
no, not necessary: the main concern is about result of `makeBigScoreTask`

Reply:
@michalkurka imo, it's currently almost impossible to ensure a completely graceful training cancellation: I don't think we have it perfect currently. This NPE was just an exception I've seen several times in the logs when AutoML times out.
But for core concerns, it's relevant only for job cancellation, so the less exceptions, the nicer it is, but I'm sure there are still a few hiding here and there.

Reply:
yeah, that is what I meant - `protected BigScore makeBigScoreTask` - makeBigScoreTask is protected and the models can call it (most likely in their overridden predictScoreImpl or other scoring functions).

XGBoost used to call it but it doesn't anymore, I checked. Just wondering about the other places.